NitroxPatcher/Patches/Dynamic/BeaconLabel_SetLabel_Patch.cs

```diff
@@ -0,0 +1,29 @@
+using System.Reflection;
+using NitroxClient.GameLogic;
+using NitroxModel.DataStructures;
+using NitroxModel.Helper;
+
+namespace NitroxPatcher.Patches.Dynamic;
+
+public sealed partial class BeaconLabel_OnHandClick_Patch : NitroxPatch, IDynamicPatch
+{
+    private static readonly MethodInfo TARGET_METHOD = Reflect.Method((BeaconLabel t) => t.OnHandClick(default));
+
+    public static bool Prefix(BeaconLabel __instance)
```

Comment:
I thing this can easily be convertet to a Transpiler which forwards the callback to a static method with populates it with our listener, after that it returns the callback which should put it back on the stack.